### ISSUE TITLE
Day of week implementation uses wrong modulo command on bigquery integration

### ISSUE THREAD

**Issue by KennethLim314** [metabase/metabase#15064]
**Describe the bug**
Aggregation by datetime:Day of week fails, getting stuck at "Doing Science" Indefinitely. Generated SQL shows use of the "%" 
character to perform a modulo command, which is an invalid character in bigquery Standard SQL
Generated SQL is as follows
```
SELECT CASE WHEN ((extract(dayofweek from `integrated_reporting_precomp.integrated_activity`.`event_datetime`) + -1) % 7) = 0 THEN 7 ELSE ((extract(dayofweek from `integrated_reporting_precomp.integrated_activity`.`event_datetime`) + -1) % 7) END AS `event_datetime`, count(*) AS `count`
FROM `integrated_reporting_precomp.integrated_activity`
GROUP BY `event_datetime`
ORDER BY `event_datetime` ASC
```

**Logs**
N/A

**To Reproduce**
Pick a given table
Select COUNT, then choose DAY OF WEEK as the aggregation criteria

**Expected behavior**
Aggregation occurs, using the day of week

**Screenshots**
![image](https://user-images.githubusercontent.com/47406716/109947926-ce5dbe00-7d14-11eb-9b3a-b1e34e7dedfb.png)

**Information about your Metabase Installation:**
```
{
  "browser-info": {
    "language": "en-GB",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.182 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.10+9",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.10",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.10+9",
    "os.name": "Linux",
    "os.version": "4.14.186-146.268.amzn2.x86_64",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "bigquery",
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "11.7"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.18"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2021-02-19",
      "tag": "v1.38.0.1",
      "branch": "release-x.38.x",
      "hash": "0635914"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```
**Severity**
Moderate. All viz using day of week aggregation fails.

**Additional context**

**Comment by flamber** [comment]
Hi @KennethLim314 
Looks like a duplicate of #15044, which was fixed yesterday and is part of 1.38.1

**Comment by Klim314** [comment]
Ah, it does appear to be a duplicate. Apologies then.
